README.md

```diff
@@ -119,48 +127,50 @@ db = MySQLDatabase(
     **connect_kwargs,
 )
 
-class TestModel(Model):
-    class Meta:
-        database = db
-        table_name = 'test'
-
+class DocumentModel(Model):
     embedding = VectorField(3)
-
-# or add hnsw index when creating table
-class TestModelWithIndex(Model):
     class Meta:
         database = db
-        table_name = 'test_with_index'
-
-    embedding = VectorField(3, constraints=[SQL("COMMENT 'hnsw(distance=l2)'")])
-
+        table_name = 'peewee_documents'
 
 db.connect()
-db.create_tables([TestModel, TestModelWithIndex])
+db.create_tables([DocumentModel])
 ```
 
 Insert vector data
 
 ```python
-TestModel.create(embedding=[1, 2, 3])
+DocumentModel.create(embedding=[1, 2, 3])
 ```
 
 Get the nearest neighbors
 
 ```python
-TestModel.select().order_by(TestModel.embedding.l2_distance([1, 2, 3.1])).limit(5)
+DocumentModel.select().order_by(DocumentModel.embedding.l2_distance([1, 2, 3.1])).limit(5)
 ```
 
 Get the distance
 
 ```python
-TestModel.select(TestModel.embedding.cosine_distance([1, 2, 3.1]).alias('distance'))
+DocumentModel.select(DocumentModel.embedding.cosine_distance([1, 2, 3.1]).alias('distance'))
 ```
 
 Get within a certain distance
 
 ```python
-TestModel.select().where(TestModel.embedding.l2_distance([1, 2, 3.1]) < 0.5)
+DocumentModel.select().where(DocumentModel.embedding.l2_distance([1, 2, 3.1]) < 0.5)
+```
+
+Add hnsw index
+
+```python
+# vector index currently depends on tiflash
+db.execute_sql(SQL(
+    "ALTER TABLE peewee_documents SET TIFLASH REPLICA 1;"
+))
+DocumentModel.add_index(SQL(
+    "CREATE VECTOR INDEX idx_embedding ON peewee_documents ((vec_cosine_distance(embedding))) USING HNSW"
+))
```

Comment:
Can we avoid explicit SQL? We can follow how pgvector does: https://github.com/pgvector/pgvector-python?tab=readme-ov-file#peewee

Reply:
Peewee does not support the `vector` prefix for defining an index,, so that we still need to use raw sql. https://github.com/coleifer/peewee/blob/master/peewee.py#L2964

Reply:
Nice digging! Looks like it is indeed very hard. When you are free, could you try with something like our own derived Index? Like: 

```python
class VectorIndex(peewee.Index):
    ...
```